deploy-extensions docs updated to reflect the latest changes (#8180)
auth0/docs: articles/extensions/bitbucket-deploy.md

* **USER**: The username used to access the Bitbucket account. Make sure you use the username, and not the email
* **PASSWORD**: The user password or an app password you create through the Bitbucket settings to grant permissions to certain apps (`Repositories: Read` permission is required)
* **BASE_DIR**: The base directory, where all your tenant settings are stored

> * **AUTO_REDEPLOY**: If enabled, the extension will try to re-deploy last successful configuration in case of failed deployment. Manual deployments and validation errors does not trigger auto-redeployment

Reviewer: ```suggestion
* **AUTO_REDEPLOY**: If enabled, the extension redeploys the last successful configuration in the event of a deployment failure. Manual deployments and validation errors does not trigger auto-redeployment
```

}
```

## Excluded records

> You can exclude `rules`, `clients`, `databases`, `connections` and `resourceServers` records from the deployment process. Excluded records won't be affected by deployment in any way.

Reviewer: ```suggestion
You can exclude the following records from the deployment process: `rules`, `clients`, `databases`, `connections` and `resourceServers`. If excluded, the records will not be modified by deployments.
```

![](/media/articles/extensions/deploy-extensions/excluded-rules.png)

## Keywords Mapping

Beginning with version **3.0.0**, you can use keywords mapping to manage your secrets and tenant-based environment variables.

> There are two ways to use the keyword mappings: You can either wrap the key in `@@key@@` or `##key##`. 

Reviewer: ```suggestion
There are two ways to use the keyword mappings. You can either wrap the key using `@` symbols (e.g., `@@key@@`), or you can wrap the key using `#` symbols (e.g., `##key##`). 
```

>   - If you use the `@` symbols, it will do a `JSON.stringify` on your value before replacing it.  So if it is a string, it will add quotes. and if it is an array or object, it will add braces.  

Reviewer: ```suggestion
  - If you use `@` symbols, your value will be converted from a JavaScript object or value to a JSON string.
```

>   - If you use the `#` symbol instead, it will just do a literal replacement; it will not add quotes or brackets.

Reviewer: ```suggestion
  - If you use `#` symbols, Auth0 will perform a literal replacement.
```

> For example, you could specify a different JWT timeout in your dev environment, and then use prod for testing and a different environment URL. 

Reviewer: ```suggestion
This is useful for something like specifying different variables across your environments. For example, you could specify different JWT timeouts for your Development, QA/Testing, and Production environments.

Refer to the snippets below for sample implementations:
```

> See the examples below.

Reviewer: ```suggestion
```
